File: 3-Bimestre/Python/numero-primo.py
"""
Aluno: Carlos Eduardo Félix
3º Bimestre
Exercício 08 - Número primo
"""

numero = int(input("Digite um número inteiro positivo: "))
divisor = 1
quantidade_divisores = 0

if numero > 0:
    while divisor <= numero:
        if numero % divisor == 0:
            quantidade_divisores += 1
        divisor += 1

    if quantidade_divisores == 2:
        print("O número é primo.")
    else:
        print("O número não é primo.")
else:
    print("Número inválido.")

File: 3-Bimestre/C++/numero-primo.cpp
/*
Aluno: Carlos Eduardo Félix
3º Bimestre
Exercício 08 - Número primo
*/
#include <iostream>
using namespace std;

int main() {
    int numero;
    int divisor = 1;
    int quantidadeDivisores = 0;

    cout << "Digite um numero inteiro positivo: ";
    cin >> numero;

    if (numero <= 0) {
        cout << "Numero invalido." << endl;
        return 0;
    }

    while (divisor <= numero) {
        if (numero % divisor == 0) {
            quantidadeDivisores++;
        }
        divisor++;
    }

    if (quantidadeDivisores == 2) {
        cout << "O numero e primo." << endl;
    } else {
        cout << "O numero nao e primo." << endl;
    }

    return 0;
}
